src/common/regression_utils.h

```diff
@@ -0,0 +1,131 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_REGRESSION_UTILS_H
+#define CEPH_REGRESSION_UTILS_H
+
+#include <iostream>
+#include <vector>
+#include <cmath>
+#include <boost/numeric/ublas/matrix.hpp>
+
+#define Z_P 2.33  // z score for 99th percentile
+
+
+namespace ceph {
+  /***
+  * Calculate the inverse of a 2x2 matrix.
+  * @param matrix<double>& m, an square 2x2 matrix
+  * @return the inverse of the m (m^-1)
+  */
+  static boost::numeric::ublas::matrix<double>
+  matrix_inverse(boost::numeric::ublas::matrix<double> &m) {
+    assert(m.size1() == m.size2() &&
+           "Can only calculate the inverse of square matrices");
+    assert(m.size1() == 2 && m.size2() == 2 && "Only for 2x2 matrices");
+    boost::numeric::ublas::matrix<double> m_inverse(2, 2);
+    const double a = m(0, 0);
+    const double b = m(0, 1);
+    const double c = m(1, 0);
+    const double d = m(1, 1);
+    const double determinant = 1.0 / ((a * d) - (b * c));
+    m_inverse(0, 0) = d * determinant;
+    m_inverse(0, 1) = -b * determinant;
+    m_inverse(1, 0) = -c * determinant;
+    m_inverse(1, 1) = a * determinant;
+    return m_inverse;
+  }
+
+  /***
+  * Find a logarithmic function in form of "y = a + b * ln(x)" which fits
+  * the given points (x_values and y_values).
+  * @param std::vector<double> x_values, x values for sample points
+  * @param std::vector<double> y_values, y values for sample points
+  * @param double theta[2], holds the a and b as output (theta[0] = a and theta[1] = b)
+  */
+  static void logarithmic_regression(
+    const std::vector<double> &x_values,
```

Comment:
I think this is a linear regression, but applied on values that were logarithmed.
I think we should modify function name.

https://www.stat.purdue.edu/~boli/stat512/lectures/topic3.pdf